src/runtime/recipe/particle.ts

```diff
@@ -245,14 +246,68 @@ export class Particle implements Comparable<Particle> {
     return this._connections[name];
   }
 
-  allConnections() {
+  allConnections(): HandleConnection[] {
     return Object.values(this._connections).concat(this._unnamedConnections);
   }
 
   ensureConnectionName(name: string) {
     return this._connections[name] || this.addConnectionName(name);
   }
 
+  getSlotConnectionNames(): string[] {
+    return Object.keys(this._consumedSlotConnections);
+  }
+
+  getSlandleConnectionByName(name: string): SlotConnection {
+    if (name in this._connections) {
+      const slandle: SlotConnection = this._connections[name].toSlotConnection();
+      return slandle;
+    }
+    return this._consumedSlotConnections[name];
+  }
+
+  getSlotConnectionByName(name: string): SlotConnection {
+    return this._consumedSlotConnections[name];
+  }
+
+  getSlotConnectionBySpec(spec: ConsumeSlotConnectionSpec): SlotConnection {
+    return this.getSlotConnections().find(slotConn => slotConn.getSlotSpec() === spec);
+  }
+
+  getSlandleConnections(): SlotConnection[] {
+    // TODO
+    return [...Object.values(this._consumedSlotConnections), ...this.allConnections().map(conn => conn.toSlotConnection()).filter(conn => conn)];
+  }
+
+  getSlotConnections(): SlotConnection[] {
+    // TODO
+    return Object.values(this._consumedSlotConnections);
+  }
+
+  getSlotSpecByName(name: string) : ConsumeSlotConnectionSpec|undefined {
+    if (!this.spec) return undefined;
+    const slot = this.spec.slotConnections.get(name);
+    if (slot) return slot;
+
+    // TODO(jopra): Provided slots should always be listed in the particle spec.
```

Comment:
What is the work item here? I don't understand the todo. With slandels all slot connections will be handle connections. It is reasonably obvious that handle connections should be in the particle spec.

Reply:
In the light of how slandles has progressed I think this todo can be removed.
The work item was to simplify the representation of slot connections so that all slot connections were visible at the particle level, without having to traverse the consumed slot connections of a particle.